openprescribing/matrixstore/build/download_practice_stats.py

```diff
@@ -28,19 +28,37 @@ def ensure_stats_downloaded_for_date(date):
     filename = get_practice_stats_filename(date)
     if os.path.exists(filename):
         return
+    client = Client("hscic")
+    check_stats_in_bigquery(date, client)
     logger.info("Downloading practice statistics for %s", date)
     temp_name = get_temp_filename(filename)
-    result = Client("hscic").query(
+    result = client.query(
         """
         SELECT *
         FROM {hscic}.practice_statistics_all_years
         WHERE month = TIMESTAMP("%s")
         """
         % (date,)
     )
+
     with gzip.open(temp_name, "wt") as f:
         writer = csv.writer(f)
         writer.writerow(result.field_names)
         for row in result.rows:
             writer.writerow(row)
```

Comment:
Couldn't we just do:
```py
row = None
for row in result.rows:
    writer.writerow(row)
if row is None:
   raise ValueError("No practice statistics found")
```
That feels less error prone, and less work, as we're not duplicating the query.